### DISCUSSION TITLE
nginx.pid is "" for Version 1.19.5

### DISCUSSION THREAD

**Discussion by daBee** [Homebrew/discussions#188] [Everyday usage]
I can't seem to get `nginx` to take a proper pid.  It's on my workstation.  Reinstalled it, used `--build-from-source` as well.  

- checked `nginx.conf`, reports good
- restarted `nginx` several times, still the same
- deleted `/usr/local/var/run/nginx.pid` several times, came back still as ""
- logged out, and back in, same issue

Any suggestions?

**Comment by gromgit** [comment]
What does your nginx error log say?

**Comment by daBee** [comment]
2020/11/30 19:34:51 [emerg] 73837#0: bind() to 0.0.0.0:80 failed (13: Permission denied)

**Reply by SMillerDev** [reply]
You're trying to run nginx on a port that's only available for root users as the formula explains:
> The default port has been set in $(brew --prefix)/etc/nginx/nginx.conf to 8080 so that
nginx can run without sudo.

https://formulae.brew.sh/formula/nginx

You either need to change the port or run nginx as root.

**Comment by daBee** [comment]
I've been running this port for over 10 years.  Why did this change all of a sudden?  This is my workstation.  nginx is always run as root.  That discussion has been beaten into the ground years ago.  Why did this package change?  I've been using 80, 443, and 8080 literally for years.  This shouldn't all of a sudden be non-permissable.

**Comment by daBee** [comment]
OK, uncommented this in nginx.conf:

    user  				root admin;

deleted nginx.pid, reloaded nginx, same issue.  The nginx.pid is still "".

**Reply by gromgit** [reply]
Are you sure nothing else is listening on port 80? Try running the following:
```
netstat -anv | awk '$4 ~ /\.80$/ { print $9 }'
```
The output is the PID of whatever's currently listening on port 80.

**Reply by daBee** [reply]
Nil.

**Reply by daBee** [reply]
Not sure why my post didn't take...but the answer: nil.

**Reply by gromgit** [reply]
And the output of `sudo nginx` is?

**Reply by daBee** [reply]
Same.  nil

**Comment by daBee** [comment]
Tried another reload and it works now.  Commented out `user root admin;` and reloaded.  It all works now.  

Strange days.
